core/src/main/java/org/elasticsearch/discovery/zen/ZenDiscovery.java

```diff
@@ -847,7 +846,12 @@ void handleJoinRequest(final DiscoveryNode node, final MembershipAction.JoinCall
 
             // validate the join request, will throw a failure if it fails, which will get back to the
             // node calling the join request
-            membership.sendValidateJoinRequestBlocking(node, joinTimeout);
+            try {
+                membership.sendValidateJoinRequestBlocking(node, state, joinTimeout);
+            } catch (Throwable e) {
+                callback.onFailure(new IllegalStateException("failure when sending a validation request to node", e));
```

Comment:
can we log a warn here - "failed to validate incoming join request from node [{}]" ? I think this should be visible.